app-text/aobook/aobook-1.0.3.recipe

```diff
@@ -4,47 +4,58 @@ of Japanese-language literature."
 HOMEPAGE="https://osdn.net/projects/aobook/"
 COPYRIGHT="2014-2017 Azel"
 LICENSE="BSD (3-clause)"
-REVISION="2"
+REVISION="3"
 localRevision="3"
 SOURCE_URI="https://github.com/hanya/aobook-haiku/archive/v$portVersion-$localRevision.tar.gz"
 CHECKSUM_SHA256="9b39981a204e6094563c2435979da4eea1aa2ef3d1a5e9580961a4e9812d34a7"
 SOURCE_DIR="aobook-haiku-$portVersion-$localRevision"
 
 ARCHITECTURES="!x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
 
 PROVIDES="
-	aobook = $portVersion
+	aobook$secondaryArchSuffix = $portVersion
 	app:Aobook = $portVersion
 	"
 REQUIRES="
-	haiku
-	lib:libfontconfig
-	lib:libfreetype
-	lib:libiconv
-	lib:libjpeg
-	lib:libpng
-	lib:libz
+	haiku$secondaryArchSuffix
+	lib:libfontconfig$secondaryArchSuffix
+	lib:libfreetype$secondaryArchSuffix
+	lib:libiconv$secondaryArchSuffix
+	lib:libjpeg$secondaryArchSuffix
+	lib:libpng$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
 	"
 BUILD_REQUIRES="
-	haiku_devel
-	devel:libfontconfig
-	devel:libfreetype
-	devel:libiconv
-	devel:libjpeg
-	devel:libpng
-	devel:libz
+	haiku${secondaryArchSuffix}_devel
+	devel:libfontconfig$secondaryArchSuffix
+	devel:libfreetype$secondaryArchSuffix
+	devel:libiconv$secondaryArchSuffix
+	devel:libjpeg$secondaryArchSuffix
+	devel:libpng$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
 	"
 BUILD_PREREQUIRES="
 	makefile_engine
-	cmd:g++
-	cmd:gcc
+	cmd:g++$secondaryArchSuffix
+	cmd:gcc$secondaryArchSuffix
 	cmd:make
+	cmd:sed
 	"
 
 USER_SETTINGS_FILES="
 	settings/aobook directory
 	"
 
+PATCH()
+{
+if [ "$effectiveTargetArchitecture" = x86 ]; then
```

Comment:
I guess `PATCH`
1) is only needed on x86 secondary arch
2) changes nothing on x86 primary arch

So, if I'm not mistaken, you can:
```diff
-if [ "$effectiveTargetArchitecture" = x86 ]; then
+if [ -n "$secondaryArchSuffix" -a "$effectiveTargetArchitecture" = x86 ]; then
 	FreeTypeInc="`findpaths -e -a x86 B_FIND_PATH_HEADERS_DIRECTORY freetype2`"
 	sed -i 's,$(shell findpaths -e B_FIND_PATH_HEADERS_DIRECTORY freetype2)',$FreeTypeInc, \
 		mlib/Makefile.haiku src/Makefile.haiku
 fi
```

Alternatively, if I'm not mistaken, you could:
```diff
-if [ "$effectiveTargetArchitecture" = x86 ]; then
+if [ -n "$secondaryArchSuffix" ]; then
-	FreeTypeInc="`findpaths -e -a x86 B_FIND_PATH_HEADERS_DIRECTORY freetype2`"
+	FreeTypeInc="`findpaths -e -a "$effectiveTargetArchitecture" \
+		B_FIND_PATH_HEADERS_DIRECTORY freetype2`"
 	sed -i 's,$(shell findpaths -e B_FIND_PATH_HEADERS_DIRECTORY freetype2)',$FreeTypeInc, \
 		mlib/Makefile.haiku src/Makefile.haiku
 fi
```
Note: I did not try any of my suggestions.

Reply:
$effectiveTargetArchitecture value could be passed to the findpaths for both primary and secondary architecture.
It looks I can remove architecture check to make the recipe simple.